pkg/router/skipper_test.go

```diff
@@ -0,0 +1,102 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSkipperRouter_Reconcile(t *testing.T) {
+	assert := assert.New(t)
+	mocks := newFixture(nil)
+
+	tests := []struct {
```

Comment:
just FYI, you can also define the struct without intermediate var.
```
	for _, tt := range []struct {
		name string
                ....
	}{{
		name: "creating new canary ingress w/ default settings",
                 ...
	}, {
                 ....
	}} {
                t.Run(tt.name, func(t *testing.T) {
                 ....
		})
	}
```

Reply:
Actually I keep on reminding that point when I review, funny I forgot 😛

Reply:
thanks - implemented https://github.com/o11n/flagger/pull/5